src/cpp/src/llm_pipeline.cpp

```diff
@@ -251,24 +265,32 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
             utils::initialize_position_ids(*position_ids, attention_mask, kv_cache_len);
         }
 
+        if(m_adapter_controller) {
+            m_adapter_controller->apply(m_model_runner, config.adapters);
+        }
+
         ov::genai::EncodedResults result;
         if (config.is_greedy_decoding()) {
-            result = ov::genai::greedy_decoding(m_model_runner, input_ids, concatenated_attention_mask, 
+            result = ov::genai::greedy_decoding(m_model_runner, input_ids, concatenated_attention_mask,
                                                 config, streamer_ptr, position_ids);
             m_selected_beam = 0;
         } else if (config.is_beam_search()) {
-            std::tie(result, m_selected_beam) = beam_search(m_model_runner, input_ids, concatenated_attention_mask, 
+            std::tie(result, m_selected_beam) = beam_search(m_model_runner, input_ids, concatenated_attention_mask,
                                                             config, position_ids, m_selected_beam);
         } else if (config.is_multinomial()) {
-            result = multinominal_decoding(m_model_runner, input_ids, concatenated_attention_mask, 
+            result = multinominal_decoding(m_model_runner, input_ids, concatenated_attention_mask,
                                            config, streamer_ptr, position_ids);
             m_selected_beam = 0;
         } else {
             OPENVINO_THROW("No decoding algorithm found for provided configuration parameters.");
         }
 
         if (!is_chat_conversation) {
+            // FIXME: Reset only KV cache part of state, there is also can be LoRA applied in the states and full reset will need to reapply LoRA even if the LoRA config is not changed
             m_model_runner.reset_state();
+            if(m_adapter_controller) {
+                m_adapter_controller->force_full_apply(); // FIXME: Reset only KV cache part to avoid this call
+            }
```

Comment:
is it planned to be fixed in next PRs before 2024.5 ?

Reply:
Yes. Added explicit TODO into the list in the PR description.